tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/data_source/tfdbg2_data_source.ts

```diff
@@ -0,0 +1,43 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {HttpClient} from '@angular/common/http';
+import {Injectable, InjectionToken} from '@angular/core';
+import {Observable} from 'rxjs';
+import {DebuggerRunListing} from '../store/debugger_types';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+
+export const TFDBG2_DATA_SOURCE = new InjectionToken<Tfdbg2DataSource>(
+  'tfdbg2 data source'
+);
+
+export abstract class Tfdbg2DataSource {
+  abstract fetchRuns(): Observable<DebuggerRunListing>;
+}
+
+@Injectable()
+export class Tfdbg2HttpServerDataSource implements Tfdbg2DataSource {
+  private readonly httpPathPrefix = '/data/plugin/debugger-v2';
+
+  constructor(private http: HttpClient) {}
+
+  fetchRuns() {
+    // TODO(cais): Once the backend uses an DataProvider that unifies tfdbg and
+    // non-tfdbg plugins, switch to using `tf_backend.runStore.refresh()`.
```

Comment:
Is it referring to fact that the ngrx store does not contain the runs right now? Would it be possible to add it there if so? Maybe runs here is different than that in general sense?